app/src/main/java/com/vitorpamplona/amethyst/model/LocalCache.kt

```diff
@@ -85,7 +85,8 @@ object LocalCache {
   fun getOrCreateNote(idHex: String): Note {
     return notes[idHex] ?: run {
       val answer = Note(idHex)
-      notes.put(idHex, answer)
+      val newNotes = ConcurrentHashMap(mapOf(Pair(idHex, answer)) + notes)
+      notes = newNotes
```

Comment:
Is this change to immutable necessary? Rebuilding large maps (which is 10x heavier than an array copy) 1000s of times per second has been quite slow in entry-level phones in the past. 

That's why we use a Concurrent map for the main repository.

Reply:
For some reason, the elements were not added to the map in the previous way, when I tried to get the element by the desired id, I always returned null.  That's why I did it this way, and it solved the problem.